ddtrace/debugging/_encoding.py

```diff
@@ -0,0 +1,488 @@
+import abc
+from inspect import CO_VARARGS
+from inspect import CO_VARKEYWORDS
+from itertools import islice
+import json
+import os
+import sys
+from time import time
+from types import FrameType
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterator
+from typing import List
+from typing import Optional
+from typing import TYPE_CHECKING
+from typing import Tuple
+from typing import Type
+from typing import Union
```

Comment:
nit: we can move some of these types to the `if TYPE_CHECKING` block. Aside from that LGTM